rpcserver.go

```diff
@@ -2169,6 +2169,13 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
 			in.CommitmentType)
 	}
 
+	if len(in.Memo) > 500 {
+		return nil, fmt.Errorf("provided memo (%s) is of length %d, "+
+			"exceeds 500", in.Memo, len(in.Memo))
+	}
+	memoHex := make([]byte, hex.EncodedLen(len(in.Memo)))
+	hex.Encode(memoHex, []byte(in.Memo))
```

Comment:
Why do we use up extra space by hex encoding the memo? Can't we just use `[]byte(in.Memo)` directly in the channel?

Reply:
I'm honestly not sure what the long-term benefits of hex-encoding would be. There was a bit of discussion [here](https://github.com/lightningnetwork/lnd/issues/7639#issuecomment-1533762154). You think we should just ditch the hex-encoding/decoding?

Reply:
I don't see any benefit with the hex encoding either. Since we're not interpreting the data on the DB level and are only reading it length-delimited, there's also no potential of special characters interfering with the storage logic (which it might do if we were using SQL). So I'd say we can ditch the hex encoding.